### ISSUE TITLE
Issue running "period-detection-in-poer-law-noise"

### ISSUE THREAD

**Issue by rajcft** [wegenmat-privat/period-detection-in-power-law-noise#1]
Hi,

I am trying to make the PSD of a light curve and also want to draw significance on that. For that, I came across this PSRESP code. I have downloaded the code but not sure how to install it or how to run it. The tutorial says that I have to provide my own light curve in which I am interested but somehow I am not able to find how to provide my own light curve in this code.
Could you please help me with that.

P.S> I was running the code like- python period-in-noise.py' but it is not working. I mean it is not asking for any input light curve file.

Thanks in advance,

Raj

**Comment by wegenmat** [comment]
Hi Raj,

sorry for the late reply - this repo was only an unfinished part-part-part-...-time project of mine and isn't maintained any longer.
But feel free to use and develop this code.

Anyway, if you are interested in the PSRESP method only, than you should use [this](https://github.com/wegenmat-privat/psresp).
There, you can check out [`test_psresp.py`](https://github.com/wegenmat-privat/psresp/blob/master/test_psresp.py) to see how to run the code.
This repo here is for period detection in power law light curves, while PSRESP is for analysing the power law noise of a light curve.

Basically, you download `psresp.py`, write a second script (with your light curve in it), add `from psresp import psresp` and call `psresp(t, y, dy, ...)` according to the documentation in the script.

Best
Matthias

**Comment by rajcft** [comment]
Hi,

So following your suggestion I have this script where I read my data and
pass it to the psresp. Now I got the error mentioned below

(gammapy-0.18.2) bozena@bozena:~/Documents/Software/psresp-master/new-by-me$
python mycode.py
/home/bozena/Documents/Software/psresp-master/new-by-me/psresp.py:167:
RuntimeWarning: divide by zero encountered in log10
  obs_nu = np.log10(obs_nu)
/home/bozena/Documents/Software/psresp-master/new-by-me/psresp.py:26:
RuntimeWarning: divide by zero encountered in divide
  simfreq = np.linspace(1, nt/2-1, num=nt/2, dtype='float64') / (dt*nt)
/home/bozena/Documents/Software/psresp-master/new-by-me/psresp.py:52:
RuntimeWarning: invalid value encountered in divide
  rate = (rate - avg) * sigma / std + mean
/home/bozena/Documents/Software/psresp-master/new-by-me/psresp.py:203:
RuntimeWarning: invalid value encountered in double_scalars
  lobin = int(duration * j / (bin/oversampling))
Traceback (most recent call last):
  File "mycode.py", line 26, in <module>
    result = psresp(t, y, dy, slopes, dt,df,percentile
=0.95,oversampling=20, number_simulations=1000)
  File "/home/bozena/Documents/Software/psresp-master/new-by-me/psresp.py",
line 319, in psresp
    _psresp_pro(t, y, dy, slopes[s], number_simulations, dt[b],
oversampling, df[f])
  File "/home/bozena/Documents/Software/psresp-master/new-by-me/psresp.py",
line 203, in _psresp_pro
    lobin = int(duration * j / (bin/oversampling))
ValueError: cannot convert float NaN to integer

On the other hand, when I run test_psresp.py, I did not get any error but
also I am not able to print the result i.e.  "print (result['slope'])",
just wanted to check if everything is working properly.

I then tried running the "test_plot_psresp.py" but again it is not
producing any figure, ideally, it should.

General comments: I believed that the method of lc simulation (timmerlc())
is already defined in psresp.py, So I do not need to define it again in
mycode.py as it is done in test_psresp.py.

Raj
